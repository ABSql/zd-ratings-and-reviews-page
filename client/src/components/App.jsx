/* eslint-disable import/extensions */
/* eslint-disable no-unused-vars */
import React, { useState, useEffect } from 'react';

import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import CircularProgress from '@material-ui/core/CircularProgress';
import getReviewForProduct from '../API/GetReviewForProduct';
import getProductMetadata from '../API/GetProductMetadata';
import Ratings from './ratings/Ratings.jsx';
import Reviews from './reviews/Reviews.jsx';
import './styles.css';

const App = () => {
  const [product, setProduct] = useState(4);
  const [metadata, setMetadata] = useState([]);
  const [productReview, setProductReview] = useState([]);
  const [clickedStar, setClickedStar] = useState(null);

  useEffect(() => {
    getReviewForProduct(product)
      .then((data) => setProductReview(data));
    getProductMetadata(product)
      .then((data) => setMetadata(data));
  }, []);

  const handleClickedStar = (starRating) => {
    setClickedStar(starRating);
  };
  if (metadata.ratings === undefined || productReview.results.length === 0) {
    return (
      <CircularProgress />
    );
  }
  return (
    <body className="zd-body">
      <Container>
        <div className="my-5">
          <Col xs={12}>
            <h4 className="primary">Ratings & Reviews</h4>
          </Col>
        </div>
        <Row>
          <Col sm={12} md={6} lg={4}>
            <div>
              <Col>
                <Ratings ratingsData={metadata} handleClickedStar={handleClickedStar} />
              </Col>
              <br />
            </div>
            <br />
          </Col>
          <Col sm={12} md={6} lg={8}>
            <Reviews
              reviewData={productReview}
              product={product}
              productSort={setProductReview}
              clickedStar={clickedStar}
            />
          </Col>
        </Row>
      </Container>
    </body>
  );
};

export default App;
