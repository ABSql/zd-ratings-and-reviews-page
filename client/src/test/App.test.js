import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../components/App.jsx';
import { shallow } from 'enzyme';

configure({ adapter: new Adapter() })


describe('App', () => {

    it('Renders the app', () => {
      const wrapper = shallow(<App />);
      expect(wrapper).toMatchSnapshot();
    });
  });