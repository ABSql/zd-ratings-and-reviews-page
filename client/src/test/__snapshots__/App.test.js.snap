// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders the app 1`] = `ShallowWrapper {}`;
